hcloud/load_balancers.go

```diff
@@ -351,3 +400,31 @@ func (l *loadBalancers) EnsureLoadBalancerDeleted(ctx context.Context, _ string,
 
 	return nil
 }
+
+// TODO: Add comment
+func checkIPModeSupport() (bool, error) {
+	config, err := config.GetConfig()
```

Comment:
I believe the unit tests currently fail because `GetConfig()` errors, which causes `EnsureLoadBalancer()` to error, which in turn causes a nil pointer dereference inside the tests. 

Is more mocks the solution, or should I just log any errors and return `false` from `checkIPModeSupport()`?

Reply:
Would appreciate some guidance on this :)

Reply:
I would prefer not to add the `sigs.k8s.io/controller-runtime` dependency. We primarily use `k8s.io/cloud-provider` and `k8s.io/client-go`.

We are already being passed a Client Builder in `cloud.Initialize()`. We can save that in the cloud struct and use it to create clients where necessary (ref: https://github.com/kubernetes/cloud-provider-aws/blob/d7e05d57709cd46297490b51ce0dd11a54dbea35/pkg/providers/v1/aws.go#L678)

I would also prefer to only check the version once on startup, but that might cause issues when the Kubernetes version is updated without restarting the Pod.

---

Regarding the unit tests, at that point you could mock the client that is being passed to `loadBalancers{}` in the test.